target/linux/ath79/generic/base-files/etc/board.d/01_leds

```diff
@@ -215,6 +215,7 @@ glinet,gl-x750)
 hak5,lan-turtle)
 	ucidef_set_led_netdev "wan" "WAN" "orange:system" "eth1"
 	;;
+meraki,mr12|\
 meraki,mr16)
 	ucidef_set_led_netdev "wan" "WAN" "green:wan" "eth0"
```

Comment:
try without this. On boards with an AR803x I have worked on, the switch chip itself controls the LED. This may or may not be true for these boards.

Reply:
No dice, the LED does not come up without this line, but does come up with.

Reply:
It also bugs me that I named this 'WAN' and not 'LAN'. These are dumb APs, there should not be a distinction.

Reply:
since there are 2 ethernet ports, you can set eth0 to WAN as default (there is some reasons why ports on the switch should not be WAN). This is what I did, but I don't think there is any formal consensus other than 1 port device must be LAN.

Reply:
I strongly doubt anyone will be using a Meraki MR12 as a gateway device, so I'd prefer to keep both ports on the same LAN-side bridge if possible.

You don't want e.g. failsafe mode to boot up and block ssh requests from the PoE port on an AP. This is a lot more important on the mostly identical MR62 which has a wired reset button which can be triggered from the other end of a cable.

Reply:
This is a matter of taste. Some people (including me) have the point that the initial setup should be close to what the vendor did (i.e. two LAN ports here), others think it might be better to have a WAN port as well from the start.

What's the vendor behavior for that LED?

Reply:
> What's the vendor behavior for that LED?

The LEDs are identical to those on the MR16. The port is marked with an [ethernet link symbol](https://cdn1.iconfinder.com/data/icons/it-components-1/24/ethernet_plug_network_sign_symbol_2-512.png) and turns on when a link is connected on the GBe port.

Reply:
Sounds like "lan" to me ...

Reply:
I agree. I misnamed the port when I did the MR16. I'll undo that mistake here for the MR12, although I don't think it changes anything other than the name of the LED in uci ... 

I might fix the MR16 in a PR later, like ---

```diff
diff --git a/target/linux/ath79/generic/base-files/etc/board.d/01_leds b/target/linux/ath79/generic/base-files/etc/board.d/01_leds
index a339f9dece..457494acd6 100755
--- a/target/linux/ath79/generic/base-files/etc/board.d/01_leds
+++ b/target/linux/ath79/generic/base-files/etc/board.d/01_leds
@@ -19,6 +19,7 @@ glinet,6416|\
 glinet,gl-ar300m-lite|\
 glinet,gl-ar300m16|\
 meraki,mr12|\
+meraki,mr16|\
 pcs,cap324|\
 tplink,cpe610-v1|\
 tplink,cpe610-v2)
@@ -216,9 +217,6 @@ glinet,gl-x750)
 hak5,lan-turtle)
        ucidef_set_led_netdev "wan" "WAN" "orange:system" "eth1"
        ;;
-meraki,mr16)
-       ucidef_set_led_netdev "wan" "WAN" "green:wan" "eth0"
-       ;;
 netgear,wnr2200-8m|\
 netgear,wnr2200-16m)
        ucidef_set_led_netdev "wan-amber" "WAN (amber)" "amber:wan" "eth0"
```

Reply:
> I agree. I misnamed the port when I did the MR16. I'll undo that mistake here for the MR12.

(To be clear I don't think having ucidef'd the LED as WAN actually changed anything other than what UCI thought the LED was named.)

Reply:
> (To be clear I don't think having ucidef'd the LED as WAN actually changed anything other than what UCI thought the LED was named.)

it also made us think it was labeled as "WAN" in letters instead of a symbol 😆

Reply:
I am guessing that the 4 wifi LEDs are for RSSI, since there is obviously not 4 radios

you probably want to consider using the script definition for `tplink,cpe210-v3`
and then editing the LED labels in DTS accordingly

Reply:
then you would add `rssileds` to the default packages in the makefile

Reply:
Changed and tested -- it works, turns on all 4 RSSI LEDs on connect and drops them as the STA (my phone) moves further away. Here's the patch from c0403 to e8232:

```diff
diff --git a/target/linux/ath79/dts/ar7242_meraki_mr12.dts b/target/linux/ath79/dts/ar7242_meraki_mr12.dts
index ec4c813e61..0d9f6724cb 100644
--- a/target/linux/ath79/dts/ar7242_meraki_mr12.dts
+++ b/target/linux/ath79/dts/ar7242_meraki_mr12.dts
@@ -25,28 +25,28 @@
        leds {
                compatible = "gpio-leds";
 
-               wifi1 {
-                       label = "green:wifi1";
+               link1 {
+                       label = "green:link1";
                        gpios = <&gpio 11 GPIO_ACTIVE_LOW>;
                };
 
-               wifi2 {
-                       label = "green:wifi2";
+               link2 {
+                       label = "green:link2";
                        gpios = <&gpio 12 GPIO_ACTIVE_LOW>;
                };
 
-               wifi3 {
-                       label = "green:wifi3";
+               link3 {
+                       label = "green:link3";
                        gpios = <&gpio 13 GPIO_ACTIVE_LOW>;
                };
 
-               wifi4 {
-                       label = "green:wifi4";
+               link4 {
+                       label = "green:link4";
                        gpios = <&gpio 14 GPIO_ACTIVE_LOW>;
                };
 
-               wan {
-                       label = "green:wan";
+               lan {
+                       label = "green:lan";
                        gpios = <&gpio 15 GPIO_ACTIVE_LOW>;
                };
 
diff --git a/target/linux/ath79/generic/base-files/etc/board.d/01_leds b/target/linux/ath79/generic/base-files/etc/board.d/01_leds
index a339f9dece..a4fbebcf4e 100755
--- a/target/linux/ath79/generic/base-files/etc/board.d/01_leds
+++ b/target/linux/ath79/generic/base-files/etc/board.d/01_leds
@@ -18,7 +18,6 @@ glinet,6408|\
 glinet,6416|\
 glinet,gl-ar300m-lite|\
 glinet,gl-ar300m16|\
-meraki,mr12|\
 pcs,cap324|\
 tplink,cpe610-v1|\
 tplink,cpe610-v2)
@@ -300,6 +299,7 @@ tplink,wbs510-v2)
        ucidef_set_led_rssi "rssimediumhigh" "RSSIMEDIUMHIGH" "green:link3" "wlan0" "60" "100"
        ucidef_set_led_rssi "rssihigh" "RSSIHIGH" "green:link4" "wlan0" "80" "100"
        ;;
+meraki,mr12|\
 tplink,cpe210-v2|\
 tplink,cpe210-v3)
        ucidef_set_led_netdev "lan" "LAN" "green:lan" "eth0"
diff --git a/target/linux/ath79/image/generic.mk b/target/linux/ath79/image/generic.mk
index 45d4f2a050..c32d27d10e 100644
--- a/target/linux/ath79/image/generic.mk
+++ b/target/linux/ath79/image/generic.mk
@@ -1185,7 +1185,7 @@ define Device/meraki_mr12
   DEVICE_VENDOR := Meraki
   DEVICE_MODEL := MR12
   IMAGE_SIZE := 15616k
-  DEVICE_PACKAGES := kmod-owl-loader
+  DEVICE_PACKAGES := kmod-owl-loader rssileds
   SUPPORTED_DEVICES += mr12
   DEVICE_COMPAT_VERSION := 2.0
   DEVICE_COMPAT_MESSAGE := Partitions differ from ar71xx version of MR12. Image format is incompatible. \
```